pkg/daemon/ceph/cleanup/disk.go

```diff
@@ -174,32 +174,62 @@ func (s *DiskSanitizer) buildShredArgs(disk string) []string {
 	return shredArgs
 }
 
+func (s *DiskSanitizer) buildQuickShredCommands(disk string) []ShredCommand {
+	return []ShredCommand{
+		{command: "ceph-volume", args: []string{"lvm", "zap", disk}},
+	}
+}
+
+func (s *DiskSanitizer) buildShredCommands(disk string) []ShredCommand {
+	var shredCommands []ShredCommand
+
+	if s.sanitizeDisksSpec.Method == cephv1.SanitizeMethodQuick {
+		return s.buildQuickShredCommands(disk)
+	}
+
+	if s.sanitizeDisksSpec.DataSource == cephv1.SanitizeDataSourceZero {
+		shredCommands = append(shredCommands, ShredCommand{command: completeShredUtility, args: s.buildShredArgs(disk)})
+		return shredCommands
+	}
+
+	shredCommands = append(shredCommands, ShredCommand{command: completeShredUtility, args: s.buildShredArgs(disk)})
+
+	return shredCommands
+}
+
 func (s *DiskSanitizer) executeSanitizeCommand(osdInfo oposd.OSDInfo, wg *sync.WaitGroup) {
 	// On return, notify the WaitGroup that we’re done
 	defer wg.Done()
 
+	// If the device is encrypted, get the real path and remove the dm device
+	if osdInfo.Encrypted {
+		realPath, err := osd.GetBackingDeviceForEncryptedBlock(s.context, osdInfo.BlockPath)
+		if err != nil {
+			logger.Errorf("failed to get backing device for encrypted block %q. %v", osdInfo.BlockPath, err)
```

Comment:
`realpath` won't be available if there any error.  Should we continue after the error?